### DISCUSSION TITLE
How much each index type consume memory?  I can't find a table

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [milvus-io/milvus#36327] [Q&A and General discussion] [answered]
I forgot the link to the table which shows how much memory each index type consumes.
Can someone please share it one again?

**Comment by yhmo** [comment, marked as answer]
The theoretical in-memory size of each index type:
![index_size](https://github.com/user-attachments/assets/8b2ddf50-8cc7-44a1-88be-58ccd7b43020)

d is dimension
nb is the number of vectors

Each dimension of vector is represented as a float32 value, which occupies 4 bytes. So, the original data size is: row_count * dimension * 4 bytes
Typically, the index size compare to the original vectors size:
- IVF_FLAT/HNSW, a bit larger than the original vectors size
- IVF_SQ8/IVF_PQ, 25% ~ 30% of the original vectors size, the recall rate of IVF_PQ is not good.
- DISKANN, 1/4 ~ 1/6 of the original vectors size, performance is not good, requires high-performance hard disk.
